common/app/views/support/HtmlCleaner.scala

```diff
@@ -341,11 +341,12 @@ case class TagLinker(article: Article)(implicit val edition: Edition, implicit v
 
     if (article.content.showInRelated) {
 
-      // Get all paragraphs which are not contained in a pullquote
-      val paragraphs = doc.getElementsByTag("p").filterNot( p =>
-        p.parents.exists( ancestor =>
-          ancestor.tagName() == "aside" && ancestor.hasClass("element-pullquote")
-        )
+      // Get all paragraphs which are not contained in a pullquote or in an instagram caption
+      val paragraphs = doc.getElementsByTag("p").filterNot( p => p.parents.exists { ancestor =>
```

Comment:
my "rule" is work out how you would describe what you want the code to do to someone.  Then make sure that you structure the code the same way, e.g. one line per each piece you describe, and functions for the bits you describe separately.  And try to minimise the plumbing parts.